src/vue/pages/Funds.vue

```diff
@@ -1,61 +1,403 @@
+                baseHardCap: {
+                  value: fund.baseHardCap,
+                  currency: fund.baseAsset
+                },
+                hardCap: {
+                  value: fund.hardCap,
+                  currency: fund.defaultQuoteAsset
+                }
+              })"
+              :html="false"
+            />
+          </div>
+        </a>
+      </div>
+    </template>
+
+    <template v-else-if="isLoaded">
+      <no-data-message
+        icon-name="inbox"
+        :msg-title="'funds.no-funds-title' | globalize"
+        :msg-message="'funds.no-funds-desc' | globalize"
+      />
+    </template>
+
+    <template v-else>
+      <loader :message-id="'funds.loading-msg'" />
+    </template>
+
+    <collection-loader
+      v-show="filteredFunds.length"
+      class="funds__loader"
+      :first-page-loader="recordsLoader"
+      @first-page-load="setRecords"
+      @next-page-load="extendRecords"
+    />
   </div>
 </template>
 
 <script>
 import TopBar from '@/vue/common/TopBar'
 import Drawer from '@/vue/common/Drawer'
+import Loader from '@/vue/common/Loader'
+import CollectionLoader from '@/vue/common/CollectionLoader'
+import NoDataMessage from '@/vue/common/NoDataMessage'
+
+import SelectField from '@/vue/fields/SelectField'
 import CreateFundForm from '@/vue/forms/CreateFundForm'
+import FundShortDetails from '@/vue/pages/funds/FundShortDetails'
+
+import VueMarkdown from 'vue-markdown'
+
+import { Sdk } from '@/sdk'
 
-import { ACCOUNT_TYPES } from '@tokend/js-sdk'
-import { mapGetters } from 'vuex'
-import { vuexTypes } from '@/vuex'
+import { SaleRecord } from '@/js/records/entities/sale.record'
+
+import config from '@/config'
+
+const FUND_STATES = {
+  live: {
+    label: 'funds.fund-live-state',
+    value: 'live',
+  },
+  upcoming: {
+    label: 'funds.fund-upcoming-state',
+    value: 'upcoming',
+  },
+  all: {
+    label: 'funds.fund-all-state',
+    value: 'all',
+  },
+}
+
+const FUND_SORT_STATES = {
+  endingSoonest: {
+    value: '3',
+    label: 'funds.sort-by-ending-soonest',
+  },
+  popularity: {
+    value: '4',
+    label: 'funds.sort-by-popularity',
+  },
+  launchDate: {
+    value: '',
+    label: 'funds.sort-by-launch-date',
+  },
+}
+
+const ALL_TOKENS_FILTER = 'funds.all-tokens-filter'
 
 export default {
   name: 'funds',
   components: {
     TopBar,
     Drawer,
+    Loader,
+    CollectionLoader,
+    NoDataMessage,
+    SelectField,
     CreateFundForm,
+    FundShortDetails,
+    VueMarkdown,
   },
   data: _ => ({
-    ACCOUNT_TYPES,
+    fundRecords: [],
+    filters: {
+      baseAsset: ALL_TOKENS_FILTER,
+      state: FUND_STATES.live.value,
+      sort: FUND_SORT_STATES.launchDate.value,
+    },
+    isLoaded: false,
     isCreateFundDrawerShown: false,
+    isDetailsDrawerShown: false,
+    selectedFund: null,
+    config,
+    FUND_STATES,
+    FUND_SORT_STATES,
   }),
+
   computed: {
-    ...mapGetters([
-      vuexTypes.account,
-    ]),
+    fundAssets () {
```

Comment:
i suggest making the field `<input>` type instead of `<select>`. This can reduce a lot of such strange code

Reply:
Used input field